libraries/core_libs/consensus/src/rewards/block_stats.cpp

```diff
@@ -117,8 +121,13 @@ void BlockStats::processDagBlocksAspen(const PeriodData& block) {
   for (const auto& dag_block : block.dag_blocks) {
     const addr_t& dag_block_author = dag_block.getSender();
     if (dag_block.getDifficulty() == min_difficulty) {
-      validators_stats_[dag_block_author].dag_blocks_count_ += 1;
-      total_dag_blocks_count_ += 1;
+      const auto& dag_block_level = dag_block.getLevel();
+      // This check prevents rewarding dag block author which produces multiple dag blocks on the same level
+      if (!bamboo_dag_rewards || dag_block_level > dag_blocks_author_level[dag_block.getSender()]) {
+        dag_blocks_author_level[dag_block.getSender()] = dag_block_level;
+        validators_stats_[dag_block_author].dag_blocks_count_ += 1;
+        total_dag_blocks_count_ += 1;
+      }
```

Comment:
I think it will be cleaner like this
```
// do't reward block with the same a lower level 
if (bamboo_dag_rewards && dag_block_level <= dag_blocks_author_level[dag_block_author]) {
   continue;
}
```

Reply:
I will change it to this

Reply:
Actually I cannot do it this way because there is a section under if which need to run. This part:
for (const auto& tx_hash : dag_block.getTrxs()) {
      addTransaction(tx_hash, dag_block_author);
    }

Reply:
I think you can move it above, right before the if

Reply:
I moved the trx iteration above the if and rearranged it this way. Please check again